src/Symfony/Component/Asset/VersionStrategy/CachedVersionStrategy.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\VersionStrategy;
+
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+/**
+ * Cache generated versions and path of any version strategy.
+ */
+class CachedVersionStrategy implements VersionStrategyInterface
+{
+    private $strategy;
+
+    private $cache;
+
+    private $options;
+
+    /**
+     * @param array $options
+     *
+     * List of available options:
```

Comment:
what about removing the options altogether and deegating them to the cache pool instead?
using `ProxyAdapter`, this is quite easy to achieve